### ISSUE TITLE
Retire fails on certain classes with associations

### ISSUE THREAD

**Issue by bdionne** [NCIEVS/nci-protege5#358]
Associations are now coded with a range `xsd:anyURI` so that the edit panel will prompt for values using an entity editor. However there are some legacy values coded as `xsd:string` which causes a fail in the retirement code that creates `OLD_ASSOCIATION` properties.

This is fixed in the next release, just adding a ticket as it's reproducible in the current code.

````

The value of the annotation is: "http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#CTRP_Terminology"^^xsd:anyURI
The range: AnnotationPropertyRange(<http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#Concept_In_Subset> <http://www.w3.org/2001/XMLSchema#anyURI>)
[1;31mUncaught Exception in thread 'AWT-EventQueue-0'[0;39m 
java.lang.IllegalStateException: Optional.get() cannot be called on an absent value
	at com.google.common.base.Absent.get(Absent.java:47) ~[guava-18.0.jar:na]
	at gov.nih.nci.ui.NCIEditTab.completeRetire(NCIEditTab.java:714) ~[na:na]
	at gov.nih.nci.ui.RetirePanel$1.actionPerformed(RetirePanel.java:170) ~[na:na]
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022) ~[na:1.8.0_151]
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348) ~[na:1.8.0_151]
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402) ~[na:1.8.0_151]
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259) ~[na:1.8.0_151]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252) ~[na:1.8.0_151]
	at java.awt.Component.processMouseEvent(Component.java:6533) ~[na:1.8.0_151]
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324) ~[na:1.8.0_151]
	at java.awt.Component.processEvent(Component.java:6298) ~[na:1.8.0_151]
	at java.awt.Container.processEvent(Container.java:2236) ~[na:1.8.0_151]
	at java.awt.Component.dispatchEventImpl(Component.java:4889) ~[na:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Container.java:2294) ~[na:1.8.0_151]
	at java.awt.Component.dispatchEvent(Component.java:4711) ~[na:1.8.0_151]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888) ~[na:1.8.0_151]
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525) ~[na:1.8.0_151]
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466) ~[na:1.8.0_151]
	at java.awt.Container.dispatchEventImpl(Container.java:2280) ~[na:1.8.0_151]
	at java.awt.Window.dispatchEventImpl(Window.java:2746) ~[na:1.8.0_151]
	at java.awt.Component.dispatchEvent(Component.java:4711) ~[na:1.8.0_151]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758) ~[na:1.8.0_151]
	at java.awt.EventQueue.access$500(EventQueue.java:97) ~[na:1.8.0_151]
	at java.awt.EventQueue$3.run(EventQueue.java:709) ~[na:1.8.0_151]
	at java.awt.EventQueue$3.run(EventQueue.java:703) ~[na:1.8.0_151]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_151]
````

**Comment by bdionne** [comment]
Itendra: How to associate classes with old associations which were failing earlier when retired?

This may not be an issue in current production at all. It's one we found locally in dev. The issue is that legacy associations were coded with a range of `xsd:string` but now are coded as `xsd:anyURI` as noted above. 

So you could test this by simply adding a new association with `xsd:string` and seeing if it behave properly on retirement.

**Comment by singhik** [comment]
Hi,
I have successfully retested this fix in QA tier V2.2.0 .  Please see attached screen shots for more details.  
[Screen_Prints.docx](https://github.com/NCIEVS/nci-protege5/files/1732674/Screen_Prints.docx)

Unable to change the labels as don't have access, this ticket now is Pass.

Regards,
Itendra Singh
